lib/dradis/plugins/template_service.rb

```diff
@@ -90,11 +83,55 @@ def template_source
 
       private
 
+      def apply_mapping(mapping)
+        # fetch mapping_fields through mapping or default mapping fields through plugin
+        mapping_fields =
+          if mapping && mapping.mapping_fields.any?
+            mapping.mapping_fields
+          else
+            plugin::Mapping.default_mapping[template]
```

Comment:
I'm not convinced this is the right structure. If you have a separate #mapping method, then that method should either return the mapping, or return the default. The rest of the code shouldn't constantly be checking for `mapping == nil`